### DISCUSSION TITLE
How to specify proxy?

### DISCUSSION THREAD

**Discussion by opoplawski** [Helioviewer-Project/JHelioviewer-SWHV#233] [Q&A]
It appears that jhelioviewer does not automatically discover the network/system configured proxy (which would be ideal).  There also appears to be no way to specify one in the settings.  How is one supposed to do this?  I've tried adding to user.properties:
```
jvm.arguments=-Dhttp.proxyHost\=HOSTNAME -Dhttp.proxyPort\=8080 -Dhttps.proxyHost\=HOSTNAME -Dhttps.proxyPort\=8080
```
to no effect, at least for the version check.

**Comment by bogdanni** [comment]
If you are using Linux, could you please try something like `jhelioviewer -J-Dhttp.proxyHost=HOSTNAME -J-Dhttp.proxyPort=8080 etc.`

**Comment by opoplawski** [comment]
I'm trying `/usr/lib/jvm/java-19/bin/java -jar JHelioviewer-SWHV-4.5.0/JHelioviewer.jar -Dhttp.proxyHost=HOSTNAME -Dhttp.proxyPort=8080` but that doesn't seem to have any effect at all either.

**Comment by bogdanni** [comment]
Should be more like, i.e., the -D options before the jars:
```
java --add-exports java.desktop/sun.awt=ALL-UNNAMED -Dhttp.proxyHost=HOSTNAME -Dhttp.proxyPort=8080 -cp JHelioviewer.jar:natives-linux.jar org.helioviewer.jhv.JHelioviewer
```

**Comment by opoplawski** [comment]
Ah, thanks - that seems to work on Linux.  Now how to get it to work on Windows.  Why doesn't the user.properties file work?

**Comment by bogdanni** [comment]
Try something like in the example here: http://swhv.oma.be/user_manual/#oom

The problem is that those Java system properties have to be set before JVM starts.
I will try the library you indicated, it is difficult to test without access to a real proxy server, my attempts for testing with a local one were not very conclusive, I will attempt again.

**Comment by opoplawski** [comment]
Thanks.  Looks like there is a quoting issue on the web page - need to surround the `-cp` argument with `"`'s - there is just a stray one at the end.
